kubernetes/samples/scripts/domain-lifecycle/setupPortForwarding.sh

```diff
@@ -0,0 +1,181 @@
+  WebLogic Administration Console. The 'kubectl port-forward' pattern is often more secure than
+  exposing the console to the public Internet. The script creates a pod in the domain namespace
+  using the image containing 'socat' binary and it forwards the request to the WebLogic
+  Admistration Server pod. The script derives the address and the adminstration port of the
+  WebLogic Admistration Server using the domain topology. The port can also be provided on the
+  command line using '-r' option.
+
+  Usage:
+
+    $(basename $0) -n namespace -d domain_uid [-a local_ip_address] [-l localPort] [-i image_name] 
+                   [-p image_pull_policy] [-w wait_timeout] [-p port_forward_server_name] [-m kubecli] 
+                   [-t pod_running_timeout]
+
+    -d <domain_uid>                : Domain unique-id. This parameter is required.
+
+    -n <namespace>                 : Domain namespace. This parameter is required.
+
+    -a <local_ip_address>          : Addresses to listen on (comma separated value for --address flag). 
+                                     Only accepts IP addresses or localhost as a value. Default is 'localhost'.
+
+    -l <local_port>                : A local port to listen on. If not specified, it listens on a random local port.
+
+    -r <remote_port>               : The port of the WebLogic Administartion Server to forward the request to.
+                                     If not specified, this script will derive the adminstration port from the
+                                     WebLogic domain topology stored in the introspector config map.
+
+    -i <image_name>                : Name of the image containing 'socat' binary to be used for the port-forward
+                                     server pod. Default is 'alpine/socat'.
+
+    -p <image_pull_policy>         : Image pull policy for the image to be used for port-forward server pod.
+                                     Default is 'IfNotPresent'.
+
+    -w <wait_timeout>              : Wait timeout value used when waiting for the port-forward server pod to be ready.
+
+    -s <port_forward_server_name>  : Name of the port-forward server pod created by this script. 
+                                     Default is 'port-forward-server'.
+ 
+    -t <pod_running_timeout>       : Value for --pod-running-timeout flag of kubectl port-forward command. The length of
+                                     time to wait until at least one pod is running. Default is 1m0s.
+
+    -m <kubernetes_cli>            : Kubernetes command line interface. Default is 'kubectl' if KUBERNETES_CLI env
+                                     variable is not set. Otherwise default is the value of KUBERNETES_CLI env variable.
+
+    -h                             : This help.
+   
+EOF
+exit $1
+}
+
+kubernetesCli=${KUBERNETES_CLI:-kubectl}
+domainUid=""
+domainNamespace=""
+portForwardPodName=port-forward-server
```

Comment:
Can there be more than one domain active at one time?  If so I assume the portForwardPodName should be decorated with the DOMAIN_UID.

Can there be more than one port-forward active for a domain at one time? If so, I assume it should also be decorated with the local port number...